Client.java

```diff
@@ -47,6 +52,53 @@ public void run() {
                   }
             });
 
+
+            if(feature == 'i'){
+                  System.out.println("Preforming DH to determine Integrity of both parties");
+                  DHKeyAgreement clientDH = new DHKeyAgreement();
+                  KeyPair clientKPair;// = clientDH.getClientKeyPair();
+                  byte [] serverPubKeyEnc, clientSharedSecret;
+                  int len, length;
+                  PublicKey serverPubKey;
+                  KeyAgreement clientKeyAgree;
+                  byte [] clientPubKeyEnc; 
+
+
+                  try
+                  {     
+                        clientKPair = clientDH.getClientKeyPair();
+                        clientKeyAgree = KeyAgreement.getInstance("DH");
+                        clientKeyAgree.init(clientKPair.getPrivate());
+                        clientPubKeyEnc = clientKPair.getPublic().getEncoded();
+                        len = clientPubKeyEnc.length;
+                        
+                        // Send Server ublic Key Encoded
+                        out.write(clientPubKeyEnc); 
+
+                        //Get Servers Public Key Encoded
+                        serverPubKeyEnc = new byte[100000000];
+                        int numRead = in.read(serverPubKeyEnc);
+                        System.out.println(" NUMBER BYTES READ IS " + numRead);
+
+                        //in.readFully(serverPubKeyEnc, 0, length); // Store Servers Public Key Encoded
+
+                        serverPubKey = clientDH.getDHPublicKeyFromEncoded(serverPubKeyEnc);
+                        clientKeyAgree.doPhase(serverPubKey, true);   
+
+                        clientSharedSecret = clientKeyAgree.generateSecret();
+                        // System.out.println(clientDH.toHexString(clientSharedSecret)); 
+
+                        out.flush();
+                        //out.write(clientSharedSecret.length);
+                        out.write(clientSharedSecret); 
```

Comment:
Why is this being sent back to the server? Shouldn't this not be shared over the network?

Reply:
Oh wait, is it for testing in the server file? Maybe when you do code cleanup block comment out all of your testing code with labels JIC we need it in the future.